api/src/main/java/net/kyori/adventure/audience/Audience.java

```diff
@@ -708,24 +710,192 @@ default void openBook(final @NotNull Book book) {
-  default void sendResourcePack(final @NotNull ResourcePackRequestLike request) {
-    this.sendResourcePack(request.asResourcePackRequest());
+  default void setResourcePack(final @NotNull ResourcePackCallback cb, final @NotNull ResourcePackRequestLike request, final @NotNull ResourcePackRequestLike@NotNull... others) {
+    final ResourcePackRequest[] otherReqs = new ResourcePackRequest[others.length];
+    for (int i = 0; i < others.length; i++) {
+      otherReqs[i] = others[i].asResourcePackRequest();
+    }
+    this.setResourcePack(cb, request.asResourcePackRequest(), otherReqs);
   }
 
   /**
-   * Sends a resource pack request to this audience.
+   * Sends resource pack requests to this audience, replacing any resource packs that might be present.
+   *
+   * <p>Multiple resource packs are only supported since 1.20.3. On older versions, only the first pack will be sent.</p>
    *
+   * @param cb a callback to be executed when resource pack events associated with this application are received
    * @param request the resource pack request
+   * @param others other requests
    * @see ResourcePackRequest
    * @since 4.15.0
    */
-  default void sendResourcePack(final @NotNull ResourcePackRequest request) {
+  @SuppressWarnings("checkstyle:MethodName")
+  default void setResourcePack(final @NotNull ResourcePackCallback cb, final @NotNull ResourcePackRequest request, final @NotNull ResourcePackRequest@NotNull... others) {
+  }
+
+  /**
+   * Sends resource pack requests to this audience, adding to any existing resource packs that might be present.
+   *
+   * <p>Multiple resource packs are only supported since 1.20.3. On older versions, this behaves identically to {@link #setResourcePack(ResourcePackRequestLike, ResourcePackRequestLike...)}.</p>
+   *
+   * @param request the resource pack request
+   * @param others other requests
+   * @see ResourcePackRequest
+   * @since 4.15.0
+   */
+  @ForwardingAudienceOverrideNotRequired
+  default void sendResourcePack(final @NotNull ResourcePackRequestLike request, final @NotNull ResourcePackRequestLike@NotNull... others) {
+    final ResourcePackRequest[] otherReqs = new ResourcePackRequest[others.length];
+    for (int i = 0; i < others.length; i++) {
+      otherReqs[i] = others[i].asResourcePackRequest();
+    }
+    this.sendResourcePack(request.asResourcePackRequest(), otherReqs);
+  }
+
+  /**
+   * Sends resource pack requests to this audience, adding to any existing resource packs that might be present.
+   *
+   * <p>Multiple resource packs are only supported since 1.20.3. On older versions, this behaves identically to {@link #setResourcePack(ResourcePackRequest, ResourcePackRequest...)}.</p>
+   *
+   * @param request the resource pack request
+   * @param others other requests
+   * @see ResourcePackRequest
+   * @since 4.15.0
+   */
+  default void sendResourcePack(final @NotNull ResourcePackRequest request, final @NotNull ResourcePackRequest@NotNull... others) {
```

Comment:
Can this method just delegate to the "send" methods with an empty callback? That way you don't have to implement multiple methods. You can make the empty callback some static constant that is available to implementations so they know they don't have to track it.

Reply:
eh, depending on the object identity of a no-op callback seems a bit iffy to me (unless we exposed it as API on `ResourcePackCallback` itself), but tbh as I think about it your suggestion makes more and more sense.